kernel/src/syscall/mmap.rs

```diff
@@ -57,6 +57,9 @@ fn do_sys_mmap(
     if len == 0 {
         return_errno_with_message!(Errno::EINVAL, "mmap len cannot be zero");
     }
+    if len > usize::MAX - PAGE_SIZE + 1 {
+        return_errno_with_message!(Errno::ENOMEM, "mmap len align overflow");
```

Comment:
I think there are some other overflow scenarios. e.g. when specifying `addr` or `offset`, `addr + len` or `offset + len` may be overflow, right?

Reply:
Though I did not fuzz a `offset` related overflow, but there is possibility.
We should add a check here like https://elixir.bootlin.com/linux/v6.10.5/source/mm/mmap.c#L1251
But do we have something like `PAGE_SHIFT`?

Reply:
You can feel free to add `PAGE_SHIFT`. However, it seems that Linux use `PAGE_SHIFT` is because it turns `offset` into `offset_in_pages` (i.e. pgoff). We may not need this value.